Gems/ROS2/Code/Source/Imu/ROS2ImuSensorComponent.cpp

```diff
@@ -77,25 +77,54 @@ namespace ROS2
         required.push_back(AZ_CRC_CE("ROS2Frame"));
     }
 
-    void ROS2ImuSensorComponent::SetupRefreshLoop()
+    void ROS2ImuSensorComponent::Activate()
     {
-        InstallPhysicalCallback();
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+        AZ_Assert(m_sensorConfiguration.m_publishersConfigurations.size() == 1, "Invalid configuration of publishers for IMU sensor");
+        m_imuMsg.header.frame_id = GetFrameID().c_str();
+        const auto publisherConfig = m_sensorConfiguration.m_publishersConfigurations[Internal::kImuMsgType];
+        const auto fullTopic = ROS2Names::GetNamespacedName(GetNamespace(), publisherConfig.m_topic);
+        m_imuPublisher = ros2Node->create_publisher<sensor_msgs::msg::Imu>(fullTopic.data(), publisherConfig.GetQoS());
+
+        m_linearAccelerationCovariance = ToDiagonalCovarianceMatrix(m_imuConfiguration.m_linearAccelerationVariance);
+        m_angularVelocityCovariance = ToDiagonalCovarianceMatrix(m_imuConfiguration.m_angularVelocityVariance);
+        m_orientationCovariance = ToDiagonalCovarianceMatrix(m_imuConfiguration.m_orientationVariance);
+
+        StartSensor(
+            m_sensorConfiguration.m_frequency,
+            [this](float imuDeltaTime, AzPhysics::SceneHandle sceneHandle, float physicsDeltaTime)
+            {
+                if (!m_sensorConfiguration.m_publishingEnabled)
```

Comment:
Perhaps moving condition check to the OnImuEvent makes more sense (as opposed to leaving it here for a tiny bit of optimization). But one can argue for either.

Reply:
The reason behind placing this if-statement here was to expose it to user / make it more visible. I would stick to that, to be honest.